pkg/operator/k8sutil/node.go

```diff
@@ -281,12 +281,17 @@ func GetKubernetesNodesMatchingRookNodes(ctx context.Context, rookNodes []cephv1
 	if err != nil {
 		return nodes, fmt.Errorf("failed to list kubernetes nodes. %+v", err)
 	}
-	for _, kn := range k8sNodes.Items {
-		for _, rn := range rookNodes {
+	for _, rn := range rookNodes {
+		nodeFound := false
+		for _, kn := range k8sNodes.Items {
 			if rookNodeMatchesKubernetesNode(rn, kn) {
 				nodes = append(nodes, kn)
+				nodeFound = true
 			}
 		}
+		if !nodeFound {
+			logger.Warningf("failed to find matching kubernetes node for %v. Please check CephCluster's config and make sure each 'name' field in spec.storage.nodes match their 'kubernetes.io/hostname' label", rn)
```

Comment:
small suggestions on the message
```suggestion
			logger.Warningf("failed to find matching kubernetes node for %v. Check the CephCluster's config and confirm each 'name' field in spec.storage.nodes matches their 'kubernetes.io/hostname' label", rn)
```